jobmon_client/src/jobmon/client/array.py

```diff
@@ -262,7 +263,7 @@ def create_tasks(
                 node=node,
                 task_args=self.task_args,
                 op_args=self.op_args,
-                resource_scales=resource_scales,
+                resource_scales=copy.deepcopy(resource_scales),
```

Comment:
The `copy.deepcopy` calls are for the `Iterator`, since we call `next` on it repeatedly. We need each task to have its own `Iterator`, so we don't go through the `Iterator` faster than intended.

Reply:
How many times to you deepcopy the data? It looks like you do it in a few places which seems unnecessary, once should be sufficient. Extra deepcopies don't seem very useful

Also, do you want to restrict this copy behavior to iterators only?